lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -707,7 +707,11 @@ public function computeChangeSet(ClassMetadata $class, $entity)
                 continue;
             }
 
-            $this->computeAssociationChanges($assoc, $val);
+            try {
+                $this->computeAssociationChanges($assoc, $val);
+            } catch (\Exception $ex) {
```

Comment:
@flip111 writing these tests isn't that problematic if you keep the test asset as simple as possible.